src/components/partitioned_lock/src/lib.rs

```diff
@@ -36,20 +36,30 @@ impl<T, B> PartitionedRwLock<T, B>
 where
     B: BuildHasher,
 {
-    pub fn try_new<F, E>(init_fn: F, partition_bit: usize, hash_builder: B) -> Result<Self, E>
+    /// the old way to new a partitioned lock, for Compatibility
+    pub fn try_new_bit_len<F, E>(
+        init_fn: F,
+        partition_bit_len: usize,
+        hash_builder: B,
+    ) -> Result<Self, E>
     where
         F: Fn(usize) -> Result<T, E>,
     {
-        let partition_num = 1 << partition_bit;
-        let partitions = (1..partition_num)
-            .map(|_| init_fn(partition_num).map(RwLock::new))
-            .collect::<Result<Vec<RwLock<T>>, E>>()?;
+        let partition_num = 1 << partition_bit_len;
+        PartitionedRwLock::try_new_with_partition_num(init_fn, partition_num, hash_builder)
+    }
 
-        Ok(Self {
-            partitions,
-            partition_mask: partition_num - 1,
-            hash_builder,
-        })
+    /// the suggested method of new a partitioned lock
+    pub fn try_new_suggest_cap<F, E>(
```

Comment:
try_new_with_suggest_cap?

Reply:
The methods have been renamed as suggested and proper docs has been added.